src/content/docs/references/docs/arrays.md

```diff
@@ -14,84 +14,132 @@ These are declared as `<type>[<size>]`, e.g. `int[4]`. Fixed arrays are treated
 
-## Slice
+    int[4]* b = (int[4]*)&a;    // Explicit cast required
+    int[4]* b = &a;             // ERROR no implicit conversion
 
-The final type is the slice `<type>[]`  e.g. `int[]`. A slice is a view into either a fixed or variable array. Internally it is represented as a struct containing a pointer and a size. Both fixed and variable arrays may be converted into slices, and slices may be implicitly converted to pointers:
-    
-    int[4] a = { 1, 2, 3, 4};
-    int[] b = &a; // Implicit conversion is always ok.
-    int[4] c = (int[4])b; // Will copy the value of b into c.
-    int[4]* d = (int[4])a; // Equivalent to d = &a
-    b.len; // Returns 4
-    e += 1;
-    int* f = b; // Equivalent to e = &a
-    f = d; // implicit conversion ok.
+    int* c = a;     // ERROR
+
+    int* d = &a;    // Valid implicit conversion
+    int* e = b;     // Valid implicit conversion
+    int[3] f = a;   // Copy by value!
+}
+```
 
 ### Slicing arrays
 
-It's possible to use the range syntax to create slices from pointers, arrays, and other slices. This is written
-`arr[<start index>..<end index>]`, where the end index is inclusive.
-You can also use the start + len syntax: `arr[<start index> : len]`.
+It's possible to use the range syntax to create slices from pointers, arrays, and other slices.
 
-    
+This is written `arr[<start index>..<end index>]`, where the `end index` is inclusive. 
+```c3
+fn void test() 
+{
     int[5] a = { 1, 20, 50, 100, 200 };
+
     int[] b = a[0..4]; // The whole array as a slice.
-    int[] b2 = a[0:5]; // Same as above.
     int[] c = a[2..3]; // { 50, 100 }
-    int[] c2 = a[2:2]; // Same as above.
+}
+```  
 
-It's possible to omit the first and last indices of a range, and the start index for start + len.
-Omitting the start index will default it to 0, omitting the end index will set it to the last valid
-index (this is not allowed on pointers). The length cannot be omitted in start + len syntax.
+You can also use the `arr[<start index>:<slice length>]`
+```c3
+fn void test()
+{
+    int[5] a = { 1, 20, 50, 100, 200 };
+    
+    int[] b2 = a[0:5]; // { 1, 20, 50, 100, 200 } Start index 0, slice length 5
+    int[] c2 = a[1:2]; // { 50, 100 } Start index 2, slice length 2
+}
+```
+
+It’s possible to omit the first and last indices of a range:
+- `arr[..<end index>]` Omitting the start index will default it to 0
+- `arr[<start index>..]` Omitting the `end index` will set `arr.len()` (this is not allowed on pointers)
 
-The following are all equivalent:
+Equivalently with index offset `arr[:<slice length>]` you can omit the `start index` 
 
+The following are all equivalent and slice the whole array
+
+```c3
+fn void test() 
+{
     int[5] a = { 1, 20, 50, 100, 200 };
+
     int[] b = a[0..4];
     int[] c = a[..4];
     int[] d = a[0..];
     int[] e = a[..];
+    
     int[] f = a[0:5];
     int[] g = a[:5];
+}
+```
+
+You can also slice referencing the last index, where
```

Comment:
A better way would be to describe ^i as meaning `len - i`. It's both applicable to `:` and `..` syntax, but used differently. For example getting the entire slice with `^` would be `a[0..^1]` or `a[0:^0]`. This needs to be explained I think.

Reply:
Thanks I now realised my mistake with ^i == len-i that helped a lot, in latest commit